### PR TITLE
Restore code climate, jscs

### PR DESCRIPTION
**Pull request by k88hudson**
Ok, I added comments for all the actual changes I made -- everything else is quotes and style changes

r? @rlr

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-    - firefox -v
+  - cd $TRAVIS_BUILD_DIR
+  - ./node_modules/.bin/mozilla-download --branch mozilla-central --product firefox firefox
+  - export FIREFOX_PATH=$TRAVIS_BUILD_DIR/firefox/firefox
+  - export PATH=$FIREFOX_PATH:$PATH
+  - firefox -v
 
 after_script:
-    #- ./node_modules/.bin/codeclimate-test-reporter < ./logs/reports/coverage/lcov/lcov.info
+  - npm run test:codeclimate
```
Comment (k88hudson) [inline .travis.yml:18]: Restored codeclimate here

File: package.json [hunk 2]
```diff
@@ -13,13 +13,11 @@
     "build:js": "NODE_ENV=production webpack -p --optimize-dedupe",
     "build:css": "node-sass src/main.scss -o www",
     "watch:css": "npm run build:css && npm run build:css -- --source-map www/main.css.map -w -r",
-    "watch:js": "webpack-dev-server --hot --port=1944 --output-public-path www",
+    "watch:js": "webpack-dev-server --hot --port=1944 --content-base www",
```
Comment (k88hudson) [inline package.json:16]: this fixes the server starting at the wrong place

File: package.json [hunk 3]
```diff
@@ -13,13 +13,11 @@
...
     "test:watch": "karma start --no-single-run --browsers ChromeCanary",
     "test": "karma start",
-
-    "jshint": "jshint src bin",
-    "jscs": "jscs --esnext src",
-    "jscs:fix": "jscs --esnext --fix bin/ src/locale/ src/js/ src/js/lib/ src/test/ src/test/lib"
+    "test:codeclimate": "codeclimate-test-reporter < ./logs/reports/coverage/lcov/lcov.info",
+    "posttest": "awk 'FNR==1{print \"\"}1' ./logs/reports/coverage/text-summary.txt ./logs/reports/coverage/text.txt"
```
Comment (k88hudson) [inline package.json:20]: outputs coverage results

File: package.json [hunk 4]
```diff
@@ -48,14 +45,16 @@
     "handlebars": "^4.0.4",
     "html2js": "^0.2.0",
     "inject-loader": "^2.0.1",
-    "jscs": "^2.1.0",
+    "istanbul-instrumenter-loader": "^0.1.3",
```
Comment (k88hudson) [inline package.json:48]: this is for code coverage

File: webpack.config.js [hunk 5]
```diff
@@ -21,35 +21,42 @@ module.exports = {
...
-      exclude: [/node_modules/],
-      loader: 'eslint'
-    }],
+    preLoaders: [
+      {
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        loader: "eslint"
+      },
+      {
```
Comment (k88hudson) [inline webpack.config.js:41]: added jscs to webpack as preloader

File: package.json (hunk 2 above)
Comment (rlr) [inline package.json:16]: :+1:

**Comment by rlr** [comment]
LGTM R+

**Pull request merged**